mix.exs

```diff
@@ -56,6 +56,7 @@ defmodule ExAzureKeyVault.MixProject do
     [
       {:httpoison, "~> 1.2.0"},
       {:poison, "~> 4.0.0"},
+      {:dialyxir, "~> 0.5", only: [:dev], runtime: false},
```

Comment:
Could you remove this change from this PR as well? I would like to keep this project as lean as possible.

Please create an issue or a separate PR if you want to discuss more about this dependency.

Reply:
I will remove the Dialyzer dependency, but I highly recommend you look into Dialyzer: it allows you to run typechecks and could have prevented the mistakes fixed in this PR.

Reply:
I really appreciate the suggestion. I'll take that into consideration later.